### ISSUE TITLE
Monitor: Add unsubscribe link

### ISSUE THREAD

**Issue by danjjohnson** [Automattic/jetpack#6585]
Related to https://github.com/Automattic/jetpack/issues/712

We should have an 'unsubscribe' link in Monitor emails, so that users can manage their notification preferences, especially useful if they deleted or moved their site before disconnecting Jetpack.

The link should work even if the site is no longer connected.

Suggested by @beaulebens in p7pQDF-1m5-p2

**Comment by beaulebens** [comment]
Even one step further, this is the sort of link that should probably just have a token in it, and work even if you're logged out.

**Comment by macmanx2** [comment]
It would be really awesome if this Unsubscribe (or Turn Off Monitor) link did not require them to log in, for those cases where the site is gone and they are unable to log in to their WordPress.com account.

We don't want our users to feel like marking the notification emails as Spam is their only option.

**Comment by macmanx2** [comment]
We're starting to get a lot of requests about this, maybe folks were in a rush to launch sites early last year but didn't follow through and thus let them expire.

We shouldn't leave our users feeling like marking the notification emails as Spam is their only option.

**Comment by jeherve** [comment]
There is already a Manage link, and folks can unsubscribe from there:

<img width="483" alt="screen shot 2017-04-19 at 11 52 05" src="https://cloud.githubusercontent.com/assets/426388/25194126/a8cb0e92-24f6-11e7-946a-4978fc20aeb5.png">

Until we add an "Unsubscribe" link, I'd recommend folks to click on that manage link to deactivate Monitor.

**Comment by beaulebens** [comment]
@jeherve Yep, the problem with that link is that it requires you to be able to log into a WP.com account to change your settings, and I believe that in turn requires that WP.com can talk to your site properly. That means if your site is completely gone/deleted, or you have a communication error, or whatever, then you potentially cannot disable notifications in any way.

**Comment by MichaelArestad** [comment]
@beaulebens do the notifications come from WordPress.com or from the site itself? I would guess if the site was down, you would be able to turn off monitor notifications from WordPress.com.

I think we need to make the email link more prominent on these emails.

**Comment by beaulebens** [comment]
The emails come from WordPress.com/Jetpack. Yes you should be able to disable notifications from WordPress.com for sure, and you can, but only if your site is connected/working (oh the irony!). Currently the setting lives here;

`https://wordpress.com/settings/security/{site}`

But I think some exploration was done (maybe even by you :) ) to move the settings for it over to `https://wordpress.com/me/notifications` under each site.

The problems with all of the above though are:

1. they require your site to be online and connected (to get it to show up properly in WP.com), and 
2. you have to be able to log into WordPress.com to get at them, even to unsubscribe/turn the notifications off.

So, IMHO we need a solution where;

1. we expose and allow interaction with these specific settings, even if a site is offline/having connection problems (as long as it's been connected under your account, and shows up in your site list), and separately;
2. a special tokenized URL where we can send people even if they're logged out, and it will force turn off settings for the email address/account where they received the monitor notification (see below)

For #2 above, what I mean is that you'd be able to do something like;

- Monitor checks a site, sees it offline
- Monitor checks notification preferences for that site, and sees that `user1` should receive a notification
- Monitor sends  an email to `user1`, containing a special unsubscribe link
- `user1` receives email, but doesn't want them any more, so they click the link at the bottom
- `user1` is on a brand new device, from which they've never logged in to WordPress.com before
- The link they clicked takes `user1` to a URL which immediately disables notifications for `user1`, based on data contained in the URL (without requiring them to log in or anything).
- Can provide the option to "manage your preferences" or something from there, which would require `user1` to log into their WordPress.com account to continue.

**Comment by georgestephanis** [comment]
Working on this at phab diff D11065

**Comment by dmsnell** [comment]
also, how about we just stop sending those after a certain time. I can't imagine going a month of having a site down _and_ having received emails already about it _and_ still finding the email useful

**Comment by beaulebens** [comment]
> also, how about we just stop sending those after a certain time

Good call. There's a `last_status_change` in the DB that we could use and just not send (and then maybe turn off monitoring?) if it's too long ago.

**Comment by macmanx2** [comment]
Folks might be afk and not checking email, and thus we don't want to flood their inboxes. What if we adopted something like:

1. Regular emails for the first 24 hours.
2. A "your site is still down" daily email for anything past 24 hours.
3. A "your site is still down" weekly email for anything past a week.
4. Emails stop after 4 weeks.

**Comment by rickybanister** [comment]
That sounds sensible to me.

**Comment by enejb** [comment]
I have a PR that fixes this issue by making sure that we never contact sites that have been marked as disconnected. https://code.a8c.com/D15960

**Comment by enejb** [comment]
I have a PR that fixes this issue by making sure that we never contact sites that have been marked as disconnected. D15960-code

**Comment by macmanx2** [comment]
That doesn't quite fix it, I don't think. The majority of the complaints we receive are from people who canceled their hosting and domain, but never disconnected Jetpack. For example, emails like "Of course my site is down, I didn't renew it, please stop emailing me."

Every time I've gone through the steps for these users, I have also found that their account still has a Jetpack connection for the site.

**Comment by enejb** [comment]
There is a mechanism that add the `is_disconnected` sticker for a site after 30 days of not talking to us. 
Or if we to a successful call to the site 5 time but don't get a good response. 
You can also set the sticker in a site's RC. (at the very bottom)

Although not perfect, it should clear up sites that are still getting the emails from us even though we know the sites are disconnected.

**Comment by enejb** [comment]
We already have code in place that deactivates monitor for sites that are disconnected and do not have any users.

**Comment by enejb** [comment]
@macmanx2 You are right. Having to wait a month is not a great solution before we stop sending emails completely. 

I think this could work really well in parallel with @georgestephanis solution with the email links.

**Comment by kraftbj** [comment]
Having it stop after a month is better than nothing, so 👍 this idea.

**Comment by aheckler** [comment]
Another scenario to consider: if someone deletes their WP.com account altogether, we should probably disable Monitor across the board for all their sites.

**Comment by enejb** [comment]
I just got a reply ticket from a user that wasn't able to remove stop the emails after disabled the site. 
2547083-zen 

I did a bit of testing and when the user clicks the settings button is the following:
- the settings screen loads which shows the user all the security settings. They have to scroll down to the monitor settings. 
- Once there they are not able to change the monitor settings. Since the site is ofline.

I think the ideal flow here should provide the user with a choice. 
1. Disconnect your site. This is for cases where they deleted the site.
2. Disable monitor emails. (they just want less noise)
